lhotse/cut.py

```diff
@@ -37,6 +37,19 @@ class CutUtilsMixin:
     for cuts in the future and make this an ABC instead.
     """
 
+    @property
+    def trimmed_supervisions(self) -> List[SupervisionSegment]:
+        """
+        Return the supervisions in this Cut that have modified time boundaries so as not to exceed
+        the Cut's start or end.
+
+        Note that when ``cut.supervisions`` is called, the supervisions may have negative ``start``
+        values that indicate the supervision actually begins before the cut, or ``end`` values
+        that exceed the Cut's duration (it means the supervision continued in the original recording
+        after the Cut's ending).
+        """
+        return [s.trim(self.duration) for s in self.supervisions]
```

Comment:
For ASR tasks, it would only be useful to train with trimmed cuts if we knew that they were trimmed (and on which ends) and could modify the supervision FSA accordingly.

Reply:
What do you think is the right way to return that information? It would be straightforward to return negative `start_frame` and `end_frame` exceeding the time dimension in the features, but I'm not sure if that is sufficient for K2 to adjust the supervision FSA (or how it would be adjusted). I'm curious what're your thoughts on this.

Alternatively, we can put an assertion in this dataset to ensure that there are no supervisions that need to be trimmed (at least for now).

Reply:
k2 would probably insert optional entry points and exit points.  pair of bool might be enough.
But I think it's better, for now, to just asser that this doesn't happen.  It's not something I've ever done in ASR training (cutting utterance without knowing the alignment).